sys/fuchsia/futex.txt

```diff
@@ -4,6 +4,9 @@
 include <zircon/syscalls.h>
 
 # TODO: limit value_ptr space somehow to make matches more frequent.
-zx_futex_wait(value_ptr ptr[in, int32], current_value int32, new_futext_owner zx_handle, deadline zx_time)
+zx_futex_wait(value_ptr ptr[in, int32], current_value int32, new_futex_owner zx_handle, deadline zx_time)
 zx_futex_wake(value_ptr ptr[in, int32], wake_count int32)
 zx_futex_requeue(value_ptr ptr[in, int32], wake_count int32, current_value int32, requeue_ptr ptr[in, int32], requeue_count int32, new_requeue_owner zx_handle)
+zx_futex_wake_single_owner(value_ptr ptr[in, int32])
+zx_futex_requeue_single_owner(value_ptr ptr[in, int32], current_value int32, requeue_ptr ptr[in, int32], requeue_count int32, new_requeue_owner zx_handle)
```

Comment:
Re `new_requeue_owner zx_handle`, can it be just any handle? Or some specific handle type?
We have a number of subtypes for handles:
```
syzkaller$ grep "resource.*\[zx_handle" sys/fuchsia/*.txt
sys/fuchsia/channels.txt:resource zx_chan[zx_handle]
sys/fuchsia/fifos.txt:resource zx_fifo[zx_handle]
sys/fuchsia/hypervisor_guests.txt:resource zx_vcpu[zx_handle]
sys/fuchsia/log.txt:resource zx_debug_log[zx_handle]
sys/fuchsia/ports.txt:resource zx_port[zx_handle]
sys/fuchsia/sockets.txt:resource zx_socket[zx_handle]
sys/fuchsia/sys.txt:resource zx_root_resource[zx_handle]
sys/fuchsia/tasks.txt:resource zx_task[zx_handle]
sys/fuchsia/timers.txt:resource zx_timer[zx_handle]
sys/fuchsia/vmars.txt:resource zx_vmar[zx_handle]
sys/fuchsia/vmos.txt:resource zx_vmo[zx_handle]
```

Should it be `zx_task`?

Relevant for other futex syscalls.

Reply:
`new_requeue_owner` has to be a handle to a thread.